apps/condo/domains/common/utils/session.js

```diff
@@ -0,0 +1,95 @@
+const { Session } = require('express-session')
+const get = require('lodash/get')
+const uniq = require('lodash/uniq')
+
+const { nonNull } = require('@open-condo/miniapp-utils')
+
+const PAYLOAD_CONFIG = {
+    allowedOrganizations: {
+        type: 'array',
+        transforms: [_uniqItems, _nonNullItems],
+    },
+    enabledB2BPermissions: {
+        type: 'array',
+        transforms: [_uniqItems, _nonNullItems],
+    },
+}
+
+/** Gives restrictions from session. If field === true - no restrictions were made
+ *  @param {Session} session
+ *  @example
+ *  parseSession(context.req.session)
+ *  @returns {SessionData}
+ * */
+function parseSession (session) {
+    if (!(session instanceof Session)) {
+        return makeSessionData()
+    }
+    const allowedOrganizations = get(session, 'allowedOrganizations')
+    const enabledB2BPermissions = get(session, 'enabledB2BPermissions')
+    return makeSessionData({
+        allowedOrganizations,
+        enabledB2BPermissions,
+    })
+}
+
+/**
+ * @typedef SessionDataArgs
+ * @prop {string[]?} allowedOrganizations - ids
+ * @prop {string[]?} enabledB2BPermissions - keys
+ */
+
+/**
+ * @typedef SessionData
+ * @property {string[] | null} allowedOrganizations
+ * @property {string[] | null } enabledB2BPermissions
+ */
+
+/**
+ * @param {SessionDataArgs?} payload
+ * @returns {{allowedOrganizations: string[] | null, enabledB2BPermissions: string[] | null}}
+ */
+function makeSessionData (payload = {}) {
+    const parsedPayloadEntries = Object.entries(PAYLOAD_CONFIG)
+        .map(([key, config]) => {
+            let value = payload[key]
+            if (value === undefined) {
```

Comment:
if (payload.hasOwnProperty())

Reply:
it would say true for { a: undefined }.hasOwnProperty('a'). Seems opposite of what it needs to do